lib/ember-addon-main.js

```diff
@@ -169,6 +170,9 @@ module.exports = {
 
     let isProduction = process.env.EMBER_ENV === 'production';
 
+    let checker = new VersionChecker(this.parent).for('ember-source', 'npm');
```

Comment:
I suspect this is broken in second-level or deeper addons. They're looking for ember-source in their parent only, but should be look at the top level project. Same kind of bug as https://github.com/babel/ember-cli-babel/pull/396